R/corr_plot.R

```diff
@@ -18,7 +26,82 @@
 #'
 #' corr_plot(data, features = numerical_features, method = 'spearman')
 #'
-corr_plot <- function(data, features = NULL, method='pearson', plot_width=500, plot_height=400){
-
-  NULL
-}
+corr_plot <- function(data, features = NULL, method='pearson', plot_width=11, plot_height=8){
+  
+  # Tests whether input data is of data.frame type
+  if (!is.data.frame(data)) {
+    stop("Please pass in a data.frame for 'data'")
+  }
+  
+  # Tests whether input features is a vector
+  if((!is.vector(features) | !is.character(features)) & (!is.null(features))){
+    stop("Please pass in a vector for 'features'")
+  }
+  
+  # Tests whether input data has at least two numeric features
+  if (dplyr::select_if(data, is.numeric) %>% ncol() < 2) {
+    stop("'data' needs at least two numeric features")
+  }
+  
+  # Tests whether all columns in features exist in the data   
+  for (c in features){
+    if (!(c %in% colnames(data))){
+      stop("Please pass in features with column names in the data")
+    }
+  }
+  
+  # Tests whether input method is of class character
+  if(!is.character(method)){
+    stop("Please pass in a character for 'method'")
+  }
+  
+  # Tests whether input method is one of the 3 available options 
+  if(!(method %in% c("pearson", "spearman", "kendall"))){
+    stop("Please pick a correlation method: 'pearson', 'spearman' or 'kendall'")
+  }
+  
+  # Tests whether input plot width is numeric
+  if(!is.numeric(plot_width) | length(plot_width) != 1){
+    stop("Plot_width must be a number'")
+  }
+  
+  # Tests whether input plot height is numeric
+  if(!is.numeric(plot_height) | length(plot_height) != 1){
+    stop("Plot_height must be a number'")
+  }
+  
+  #Subsetting the dataframe
+  numeric_df <- dplyr::select_if(data, is.numeric)
+  if (length(features) > 0) {
+    numeric_df <- numeric_df %>% select(all_of(features))
+  }
+  
+  # Creating corr_df dataframe
+  corr_df <- as.data.frame(cor(numeric_df, method = "pearson", use = "complete.obs"))
+  corr_df <- corr_df %>%
+    dplyr::mutate("x" = (corr_df %>% rownames())) %>%
+    tidyr::pivot_longer(names_to = "y", values_to = "corr_value", cols = where(is.numeric))
+  
+  options(repr.plot.height = plot_height, repr.plot.width = plot_width)
+  
+  # Correlation plot
+  corr_plot <- (ggplot2::ggplot(data = corr_df,
+                                ggplot2::aes(x = x, y = y, fill = corr_value)) +
```

Comment:
For R CMD CHECK to pass, we need to tell it where to evaluate these unquoted names (i.e x -> "x" column in the corr_df).
ggplot2 uses ".data$"

```suggestion
                                ggplot2::aes(x = .data$x, y = .data$y, fill = .data$corr_value)) +
```